tutorials/truffle-dev-life-cycle.md

```diff
@@ -0,0 +1,538 @@
+  --8<-- 'text/faucet/faucet-list-item.md'
+- Your own endpoint and API key for Moonbeam, which you can get from one of the supported [Endpoint Providers](/builders/get-started/endpoints/){target=_blank}
+
+## Create a Truffle Project {: #create-a-truffle-project }
+
+To quickly get started with Truffle, we're going to use the [Moonbeam Truffle Box](https://github.com/PureStake/moonbeam-truffle-box){target=_blank}, which provides a boilerplate setup for developing and deploying smart contracts on Moonbeam. 
+
+The Moonbeam Truffle Box comes pre-configured for a local Moonbeam development node and Moonbase Alpha. We'll need to add support for Moonbeam so when we're ready to deploy our contracts to MainNet, we'll be all set!
+
+It also comes with a couple of plugins: the [Moonbeam Truffle plugin](https://github.com/purestake/moonbeam-truffle-plugin){target=_blank} and the [Truffle verify plugin](https://github.com/rkalis/truffle-plugin-verify){target=_blank}. The Moonbeam Truffle plugin will help us quickly get started with a local Moonbeam development node. The Truffle verify plugin will allow us to verify our smart contracts directly from within our Truffle project.
+
+Without further ado, let's create our project:
+
+1. You can either install Truffle globally or clone the [Moonbeam Truffle Box](https://github.com/PureStake/moonbeam-truffle-box){target=_blank} repository:
+
+    ```
+    npm install -g truffle
+    mkdir moonbeam-truffle-box && cd moonbeam-truffle-box
+    truffle unbox PureStake/moonbeam-truffle-box
+    ```
+
+    To avoid globally installing Truffle, you can run the following command and then access the Truffle commands by using `npx truffle <command>`:
+
+    ```
+    git clone https://github.com/PureStake/moonbeam-truffle-box
+    cd moonbeam-truffle-box
+    ```
+
+2. Install the dependencies that come with the Moonbeam Truffle Box:
+
+    ```
+    npm install
+    ```
+
+3. Open the `truffle-config.js` file, where you'll find the network configurations for a local development node and Moonbase Alpha. We'll need to add the Moonbeam configurations here:
+
+    ```
+    ...
+    moonbeam: {
+      provider: () => {
+        ...
+        return new HDWalletProvider(
+          'PRIVATE-KEY-HERE',  // Insert your private key here
+          '{{ networks.moonbeam.rpc_url }}' // Insert your RPC URL here
+        )
+      },
+      network_id: {{ networks.moonbeam.chain_id }} (hex: {{ networks.moonbeam.hex_chain_id }}),
+    },
+    ```
+
+Now we should have a Truffle project that is configured for each of the networks we'll be deploying smart contracts to in this guide.
+
+For the sake of this guide, we can remove the `MyToken.sol` contract and the associated tests that come with the project:
+
+```
+rm contracts/MyToken.sol test/test_MyToken.js
+```
+
+## Add Simple NFT Marketplace Contract {: #example-nft-marketplace-contract }
```

Comment:
Maybe create a "##" section called "## Contract Setup" in which you have "### Add Simple..." then another for "### Add NFT.." and lastly "### Compiling your Contracts"

I found it kind of weird that the compilation step was buried inside the test